virtcontainers/qemu.go

```diff
@@ -1669,7 +1669,7 @@ func calcHotplugMemMiBSize(mem uint32, memorySectionSizeMB uint32) (uint32, erro
 
 func (q *qemu) resizeVCPUs(reqVCPUs uint32) (currentVCPUs uint32, newVCPUs uint32, err error) {
 
-	currentVCPUs = q.config.NumVCPUs + uint32(len(q.state.HotpluggedVCPUs))
+	currentVCPUs = q.qemuConfig.SMP.CPUs + uint32(len(q.state.HotpluggedVCPUs))
```

Comment:
I think this justifies a new unit test to check all the "awkward" corner case values and assert the expected behaviour. So, how about wrapping this logic in a new function to make writing that test easier? Something like:

```go
func (q *qemu) calculateCurrentVCPUs() uint32 {
    // ...
}
```

Reply:
Thanks for resolving the conflict. The area of vcpu calculation has seen a few changes so I do think a unit test would help others understand the logic here. Take a look at the following for the basic template for a table-based approach:

- https://github.com/kata-containers/tests/blob/master/Unit-Test-Advice.md.

Reply:
ok, James, thanks. I will do it later